pubsub/redis.go

```diff
@@ -38,13 +42,79 @@ func NewRedisSubscriber(node *node.Node, url string, channel string) RedisSubscr
 // Start connects to Redis and subscribes to the pubsub channel
 func (s *RedisSubscriber) Start() error {
 	// Check that URL is correct first
-	_, err := url.Parse(s.url)
+	redisUrl, err := url.Parse(s.url)
 
 	if err != nil {
 		return err
 	}
 
+	var sntnl *sentinel.Sentinel
+	var password string
+
+	if s.sentinels != "" {
+		masterName := redisUrl.Hostname()
+		password, _ = redisUrl.User.Password()
+
+		s.log.Debug("Redis sentinel enabled")
+		s.log.Debugf("Redis sentinel parameters:  sentinels: %s,  masterName: %s", s.sentinels, masterName)
+		sentinels := strings.Split(s.sentinels, ",")
+		sntnl = &sentinel.Sentinel{
+			Addrs:      sentinels,
+			MasterName: masterName,
+			Dial: func(addr string) (redis.Conn, error) {
+				timeout := 500 * time.Millisecond
+
+				c, err := redis.Dial(
+					"tcp",
+					addr,
+					redis.DialConnectTimeout(timeout),
+					redis.DialReadTimeout(timeout),
+					redis.DialReadTimeout(timeout),
+				)
+				if err != nil {
+					s.log.Debugf("Failed to connect to sentinel %s", addr)
+					return nil, err
+				}
+				s.log.Debugf("Successfully connected to sentinel %s", addr)
+				return c, nil
+			},
+		}
+
+		defer sntnl.Close()
+
+		// Periodically discover new Sentinels.
+		go func() {
```

Comment:
We need to take care (=stop) of this groutine when we exiting the `Start()` method (for example, when we exhausted all the reconnect attempts).

We can use `context.WithCancel` or simply a channel + `select`.

Reply:
Thanks again. Not sure if i did that right. Could you please check https://github.com/anycable/anycable-go/pull/95/commits/c9be92e3d4b442035be3d35dd68efddfc9268825
Cheers

Reply:
LGTM 👍